plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureRepositoryPlugin.java

```diff
@@ -50,7 +50,11 @@
 
     public AzureRepositoryPlugin(Settings settings) {
         // eagerly load client settings so that secure settings are read
-        this.azureStoreService = new AzureStorageService(settings);
+        this.azureStoreService = createAzureStoreService(settings);
+    }
+
+    AzureStorageService createAzureStoreService(final Settings settings) {
```

Comment:
Maybe add a comment here, that this is package private, non-static for testing purposes